contrib/experimental/great_expectations_experimental/expectations/expect_queried_pair_column_values_to_have_diff.py

```diff
@@ -0,0 +1,142 @@
+"""
+This is an example of a Custom QueryExpectation.
+For detailed information on QueryExpectations, please see:
+    https://docs.greatexpectations.io/docs/guides/expectations/creating_custom_expectations/how_to_create_custom_query_expectations
+"""
+
+from typing import Optional, Union
+
+from great_expectations.core.expectation_configuration import ExpectationConfiguration
+from great_expectations.exceptions.exceptions import (
+    InvalidExpectationConfigurationError,
+)
+from great_expectations.execution_engine import ExecutionEngine
+from great_expectations.expectations.expectation import (
+    ExpectationValidationResult,
+    QueryExpectation,
+)
+
+
+class ExpectQueriedPairColumnValuesToHaveDiff(QueryExpectation):
```

Comment:
@austiezr May I please propose the naming change from PairColumn to ColumnPair throughout -- so as to be consistent with the naming of similar entities in the core sections of the code?  Thanks!